lib/xnvme_be_vfio_mem.c

```diff
@@ -71,6 +71,41 @@ xnvme_be_vfio_buf_vtophys(const struct xnvme_dev *XNVME_UNUSED(dev), void *XNVME
 	errno = ENOSYS;
 	return -ENOSYS;
 }
+
+int
+xnvme_be_vfio_mem_map(const struct xnvme_dev *dev, void *vaddr, size_t nbytes, uint64_t *phys)
+{
+	struct xnvme_be_vfio_state *state = (void *)dev->be.state;
+	struct vfio_container *vfio = state->ctrl->pci.dev.vfio;
+	int err;
+
+	XNVME_DEBUG("xnvme_be_vfio_mem_map(%p, %p)", dev, vaddr);
+
+	nbytes = ALIGN_UP(nbytes, __VFN_PAGESIZE);
```

Comment:
This should be changed to a check on the alignment instead. That is, we should not map more than what the user requested and if the user did not provide a properly aligned number of bytes, then we bail out. Actually, `vfio_map_dma()` will check this (that `vaddr` **and** `nbytes` are aligned) and bail out if that is not the case.